src/js/stores/ReadyStore.js

```diff
@@ -70,15 +77,25 @@ class ReadyStore extends ReduceStore {
       case 'voterPlansForVoterRetrieve':
         // console.log('ReadyStore voterPlansForVoterRetrieve, action.res:', action.res);
         if (!action.res || !action.res.success) return state;
+        voterPlansForVoterRetrieved = true;
+
         voterPlanList = action.res.voter_plan_list || [];
-        voterPlanList.forEach((oneVoterPlan) => {
-          googleCivicElectionId = oneVoterPlan.google_civic_election_id || 0;
-          allCachedVoterPlansForVoterByElectionId[googleCivicElectionId] = oneVoterPlan;
-        });
-        return {
-          ...state,
-          allCachedVoterPlansForVoterByElectionId,
-        };
+        if (voterPlanList.length) {
+          voterPlanList.forEach((oneVoterPlan) => {
+            googleCivicElectionId = oneVoterPlan.google_civic_election_id || 0;
+            allCachedVoterPlansForVoterByElectionId[googleCivicElectionId] = oneVoterPlan;
+          });
+          return {
+            ...state,
+            allCachedVoterPlansForVoterByElectionId,
+            voterPlansForVoterRetrieved,
+          };
+        } else {
```

Comment:
As extra protection against infinite loop, maybe make this `else if (voterPlansForVoterRetrieved)` for setting "voterPlansForVoterRetrieved" to true since we only need to set it true once. Then finally `return state;` for the final `else`?